### ISSUE TITLE
Bind for selects does not set selected value when using optgroups

### ISSUE THREAD

**Issue by davidkarlsson** [aspnet/AspNetCore#10648]
When you set the value of a property that is bound to a select the option with that value will not be selected if it is nested in an optgroup.

### To Reproduce
```
<select bind="@SelectValueWithoutOptGroup">
    @foreach (var i in Enumerable.Range(1, 3))
    {
        <option value="@i">@i</option>
    }
</select>

<select bind="@SelectValueWithOptGroup">
    @foreach (var i in Enumerable.Range(1, 3))
    {
        <optgroup label="@i">
            @foreach(var j in Enumerable.Range(1, 3))
            {
                <option value="@i-@j">@i-@j</option>
            }
        </optgroup>
    }
</select>

@functions {
    string SelectValueWithoutOptGroup { get; set; }
    string SelectValueWithOptGroup { get; set; }

    protected override async Task OnInitAsync()
    {
        SelectValueWithOptGroup = "2-3";
        SelectValueWithoutOptGroup = "2";
    }
}
```

### Expected behavior
Here only the select without any optgroups will have the option selected ("2" in my example) but the one with optgroups will still have the first option selected ("1-1").  I expected the option "2-3" to be selected in the select that has optgroups as well.

### Additional context
Include the output of `dotnet --info`
```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview5-011568
 Commit:    b487ff10aa

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview5-011568\

Host (useful for support):
  Version: 3.0.0-preview5-27626-15
  Commit:  61f30f5a23

.NET Core SDKs installed:
  2.1.800-preview-009677 [C:\Program Files\dotnet\sdk]
  3.0.100-preview5-011568 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview5-19227-01 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview5-27626-15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview5-27626-15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @davidkarlsson.
This may be related #9716

**Comment by davidkarlsson** [comment]
> 
> 
> Thanks for contacting us, @davidkarlsson.
> This may be related #9716

I'm not sure if that issue is related. Wouldn't that have meant that my test with the select that doesn't have any optgroups would have failed as well? I thought the bug described in #9716 only happened if you assign a default value to to the bound property and not in `OnInitAsync` like I did in my tests which is why I filed a new issue for this.

I thought this might be related to the fact that the option is not a direct parent to the select element when you are using optgroups so that the condition here: https://github.com/aspnet/AspNetCore/blob/master/src/Components/Browser.JS/src/Rendering/BrowserRenderer.ts#L333 would return false and therefore not set the value of the select?

I'm not sure if that code is related to this though because I haven't followed the code path for the bind logic.
